pkg/terraform/stages/split.go

```diff
@@ -41,17 +41,33 @@ func WithCustomDestroy(destroy DestroyFunc) StageOption {
 	}
 }
 
+// WithNormalExtractHostAddresses returns an option for specifying that a split stage should use the normal extract host addresses process.
+func WithNormalExtractHostAddresses() StageOption {
+	return WithCustomExtractHostAddresses(normalExtractHostAddresses)
+}
+
+// WithCustomExtractHostAddresses returns an option for specifying that a split stage should use a custom extract host addresses  process.
+func WithCustomExtractHostAddresses(extractHostAddresses ExtractFunc) StageOption {
+	return func(s *SplitStage) {
+		s.extractHostAddresses = extractHostAddresses
+	}
+}
+
 // SplitStage is a split stage.
 type SplitStage struct {
 	platform             string
 	name                 string
 	destroyWithBootstrap bool
 	destroy              DestroyFunc
+	extractHostAddresses ExtractFunc
 }
 
 // DestroyFunc is a function for destroying the stage.
 type DestroyFunc func(s SplitStage, directory string, extraArgs []string) error
 
+// ExtractFunc is a function for destroying the stage.
```

Comment:
Seems a copy’n’paste from the description above

Reply:
Good find and thanks for the review!